infra/Pos/Communication/PeerState.hs

```diff
@@ -68,11 +73,21 @@ newtype PeerStateHolder m a = PeerStateHolder
                 HasLoggerName, CanLog,
                 MonadFix)
 
+instance m ~ m' => MFunctor' PeerStateHolder m m' where
+    hoist' =
+      -- ahh, deriving by hand.
```

Comment:
1. Why didn't you change `PeerStateHolder` from `newtype` to `Ether.ReaderT` as it was done for everything I've seen so far?
2. Suddenly another question appeared: why can't GHC deduce type of `coerce` here? Or it's written explicitly for clarity?

(newbie questions)